main/ui/src/main/java/org/cryptomator/ui/mainwindow/VaultDetailUnlockedController.java

```diff
@@ -14,11 +18,20 @@
 
 	private final ReadOnlyObjectProperty<Vault> vault;
 	private final VaultService vaultService;
+	private final LoadingCache<Vault, VaultStatisticsComponent> vaultStatisticsWindows;
```

Comment:
I don't understand, why a loading cache is used here. In the constructor  this cache is build without any eviction strategy, hence, the value is only loaded once. If this is true, i would suggest to use [Lazy](https://dagger.dev/api/2.10/dagger/Lazy.html) class provided by dagger

Reply:
Ideally we'd use weak values and have the whole component garbage collected. I don't know if dagger can recreated gc'ed subcomponents...? Will do some testing

Reply:
Ok so recreating is not the problem here (since we're using a component builder anyway). Problem is to keep a (strong) reference to the component from somewhere else (outside the cache). Not technically difficult, but not sure where to put this reference.